sycl/plugins/unified_runtime/ur/adapters/level_zero/usm.cpp

```diff
@@ -651,42 +630,78 @@ enum umf_result_t USMMemoryProvider::free(void *Ptr, size_t Size) {
   return UMF_RESULT_SUCCESS;
 }
 
-void USMMemoryProvider::get_last_native_error(const char **ErrMsg,
-                                              int32_t *ErrCode) {
-  (void)ErrMsg;
-  *ErrCode = static_cast<int32_t>(getLastStatusRef());
+std::pair<umf_result_t, size_t> L0MemoryProvider::GetL0MinPageSize() {
+  void *Ptr = nullptr;
+  size_t MinPageSize = 0;
+
+  auto Res = allocateImpl(&Ptr, 1, 0);
+  if (Res != UR_RESULT_SUCCESS) {
+    goto err_set_status;
+  }
+
+  // Query L0 for the minimal page size.
+  Res = USMQueryPageSize(Context, Ptr, &MinPageSize);
+  if (Res != UR_RESULT_SUCCESS) {
+    goto err_set_status;
+  }
+
+  Res = USMFreeImpl(Context, Ptr);
+  if (Res != UR_RESULT_SUCCESS) {
+    goto err_set_status;
+  }
+
+  return {UMF_RESULT_SUCCESS, MinPageSize};
+
+err_set_status:
+  getLastStatusRef() = Res;
+  return {UMF_RESULT_ERROR_MEMORY_PROVIDER_SPECIFIC, MinPageSize};
 }
 
-umf_result_t USMMemoryProvider::get_min_page_size(void *Ptr, size_t *PageSize) {
-  (void)Ptr;
+umf_result_t L0MemoryProvider::get_min_page_size(void *Ptr, size_t *PageSize) {
+  std::ignore = Ptr;
```

Comment:
Can we also add implementation for case where Ptr != NULL? We can use zeMemGetAllocProperties I think

Reply:
Done.